src/reducers/arcReducer.ts

```diff
@@ -199,46 +201,56 @@ const arcReducer = (state = initialState, action: any) => {
       });
     }
 
-    case ActionTypes.ARC_STAKE_PENDING: {
-      return update(state, { daos: {
-        [payload.stake.avatarAddress] : {
-          members: {
-            [payload.stake.stakerAddress]: {
-              stakes : { [payload.stake.proposalId] : { $set : payload.stake }},
-            },
-          },
-        },
-      }});
-    }
+    case ActionTypes.ARC_STAKE: {
+      const { meta, sequence, payload } = action as StakeAction;
+      const { avatarAddress, stakerAddress, proposalId, prediction, stake } = meta;
 
-    case ActionTypes.ARC_STAKE_REJECTED: {
-      return update(state, { daos: {
-        [payload.avatarAddress] : {
-          members: {
-            [payload.stakerAddress]: {
-              stakes : { $unset : [payload.proposalId] },
+      switch (sequence) {
+        case AsyncActionSequence.Pending:
+          return update(state, { daos: {
+            [avatarAddress] : {
+              members: {
+                [stakerAddress]: {
+                  stakes : { [proposalId] : { $set : {
+                    avatarAddress,
```

Comment:
shouldn't this all be coming from the payload? and cant you use the spread operator here?

Reply:
It's all available in the `meta`. Sure we can use the speard operator, or better: `immutability-helper` :).